src/KM_Controls.pas

```diff
@@ -4927,6 +4957,36 @@ procedure TKMColumnBox.MouseMove(X,Y: Integer; Shift: TShiftState);
 end;
 
 
+procedure TKMColumnBox.DoClick(X, Y: Integer; Shift: TShiftState; Button: TMouseButton);
+var IsClickHandled: Boolean;
+begin
+  IsClickHandled := False;
+
+  if (Button = mbLeft) and Assigned(fOnCellClick) and not KMSamePoint(fMouseOverCell, INVALID_POINT) then
+    IsClickHandled := fOnCellClick(Self, fMouseOverCell.X, fMouseOverCell.Y);
+
+  //Let propagate click event only when OnClickCell did not handle it
+  if not IsClickHandled then
+    inherited DoClick(X, Y, Shift, Button);
+end;
+
+
+procedure TKMColumnBox.MouseDown(X,Y: Integer; Shift: TShiftState; Button: TMouseButton);
+var IsMouseDownHandled: Boolean;
+begin
+  inherited;
+  MouseMove(X, Y, Shift);
+end;
+
+
+procedure TKMColumnBox.MouseMove(X,Y: Integer; Shift: TShiftState);
+begin
+  inherited;
+  UpdateMouseOverPosition(X, Y);
+  ProceedMouseDown(X, Y, Shift);

```

Comment:
I don't really understand these changes. What does ProceedMouseDown do? It's only called from MouseMove so why is it called ProceedMouseDown?